src/server/game/Server/GameClient.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __GAMECLIENT_H
+#define __GAMECLIENT_H
+
+#include "WorldSession.h"
+
+class Player;
+class Unit;
+
+class TC_GAME_API GameClient
+{
+    public:
+        GameClient(WorldSession* sessionToServer);
+
+        void AddAllowedMover(Unit* unit);
+        void RemoveAllowedMover(Unit* unit);
+        bool IsAllowedToMove(Unit* unit) const;
+        bool IsAllowedToMove(ObjectGuid guid) const;
+        void SetMovedUnit(Unit* target, bool allowMove);
+
+        Unit* GetActivelyMovedUnit() const { return _activelyMovedUnit; }
+        void SetActivelyMovedUnit(Unit* activelyMovedUnit) { _activelyMovedUnit = activelyMovedUnit; }
+
+        Player* GetBasePlayer() const { return _sessionToServer->GetPlayer(); }
+
+        void SendDirectMessage(WorldPacket const* data) const;
+    private:
+        // describe all units that this client has direct control over. Example, a player on a vehicle has client control over himself and the vehicle at the same time.
+        GuidSet _allowedMovers;
```

Comment:
why ordered set?

Reply:
Because I needed a data structure to hold guids without any duplicates.  also, i don't really care about the order on which the guids were inserted. both of these requirements make a good case for using a set.

Reply:
then how about moving into unordered? 😃

Reply:
Oh I see. Coming from Java, I thought the standard std::set was unordered mb. Thank you. I changed it.

Reply:
tbf, the asymptotic runtime improvement for a set that's going to hold single-digit number of GUIDs at best is utterly irrelevant 😛

Reply:
True of course :) It's still useful for future devs to know that the ordering doesn't matter in that data struct.